addon/compression_worker.h

```diff
@@ -0,0 +1,73 @@
+#ifndef COMPRESSION_WORKER_H
+#define COMPRESSION_WORKER_H
+#include <napi.h>
+
+#include <optional>
+#include <variant>
+
+using namespace Napi;
+
+using CompressionResult = std::variant<std::vector<uint8_t>, std::string>;
+
+// Implementation of the Overload pattern:
+// https://www.cppstories.com/2019/02/2lines3featuresoverload.html/
+template <class... Ts>
+struct overload : Ts... {
+    using Ts::operator()...;
+};
+template <class... Ts>
+overload(Ts...) -> overload<Ts...>;
+
+/**
+ * @brief An asynchronous Napi::Worker that can be with any function that produces
+ * CompressionResults.
+ * */
+class CompressionWorker : public Napi::AsyncWorker {
+   public:
+    CompressionWorker(const Napi::Env& env, std::function<CompressionResult()> worker)
+        : Napi::AsyncWorker{env, "Worker"}, m_deferred{env}, m_worker(worker), m_result{} {}
+
+    Napi::Promise GetPromise() {
+        return m_deferred.Promise();
+    }
+
+   protected:
+    void Execute() {
```

Comment:
```suggestion
    void Execute() final {
```

(not required, but best practice for sure)